src/System.IdentityModel.Tokens.Jwt/JwtConstants.cs

```diff
@@ -47,9 +47,9 @@ public static class JwtConstants
         public const string TokenTypeAlt = "urn:ietf:params:oauth:token-type:jwt";
 
         /// <summary>
-        /// Token format: 'header.payload.signature'. Signature is optional, but '.' is required.
+        /// Token format: 'header.payload.signature'. Signature is optional, and if missing, trailing '.' is also optional.
         /// </summary>
-        public const string JsonCompactSerializationRegex = @"^[A-Za-z0-9-_]+\.[A-Za-z0-9-_]+\.[A-Za-z0-9-_]*$";
+        public const string JsonCompactSerializationRegex = @"^(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)\.(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)\.?(?:[A-Za-z0-9+/]{4})*(?:[A-Za-z0-9+/]{2}==|[A-Za-z0-9+/]{3}=)?$";
```

Comment:
@tgitchel It might be a good idea to let the owners of WSO2 API Gateway know they are not generating spec compliant Compact JWT's. There could be an issue with the signature if the signature is based off of the base64 encodings.